File: 프로그래머스/기초 트레이닝 - Level 0/주사위 게임2-solve.py
def cal_score(n, a, b, c):
    return a ** n + b ** n + c ** n

def solution(a, b, c):
    answer = cal_score(1, a, b, c)
    
    if (a == b and b != c) or (b == c and a != b) or (c == a and a != b):
        answer *= cal_score(2, a, b, c)
    elif a == b and b == c: 
        answer *= cal_score(2, a, b, c) * cal_score(3, a, b, c)
        
    return answer

File: 프로그래머스/기초 트레이닝 - Level 0/주사위 게임2-solve.java
class Solution {
    public int calScore(int n, int a, int b, int c) {
        return (int) (Math.pow(a, n) + Math.pow(b, n) + Math.pow(c, n));
    }
    
    
    public int solution(int a, int b, int c) {
        int answer = calScore(1, a, b, c);
        if ((a == b && b != c) || (b == c && a != b) || (c == a && a != b)){
            answer *= calScore(2, a, b, c);
        } else if (a == b && b == c) {
            answer *= calScore(2, a, b, c) * calScore(3, a, b, c);
        } 
        
        return answer;
    }
}